File: Application Support/CotEditor/ScriptMenu/01)Dev/10)Tsv2SQLInsert.pl
#!/usr/bin/env perl

# %%%{CotEditorXInput=AllText}%%%
# %%%{CotEditorXOutput=ReplaceAllText}%%%
use strict;
use warnings;

my (@head, $val);
while (<>) {
    chomp;
    unless (@head) {
        @head = split(/\t/, $_);
    } else {
        my @values = map { 
            if (/^null$/i || !$_) {
                $val = "NULL";
            } elsif (/^[0-9]+$/) {
                $val = $_;
            } else {
                s/'/''/g;
                $val = "'" . $_ . "'";
            }
        } split(/\t/, $_);
        for my $i (($#values+1)..$#head) { push(@values, "NULL"); }
        printf("INSERT INTO xxx_table (%s) VALUES (%s);\n", join(",", @head), join(',', @values));
    }
}

File: Application Support/CotEditor/ScriptMenu/01)Dev/10)Tsv2SQLInsert.php
#!/usr/bin/php
<?php
// %%%{CotEditorXInput=Selection}%%%
// %%%{CotEditorXOutput=ReplaceSelection}%%%

require_once dirname(__FILE__) . "/-/functions.php";

$stdin = file_get_contents("php://stdin");

$tsv = Reader::tsv();
$lines = $tsv->parse($stdin);

$header = array_shift($lines);

$header_s = implode(',', $header);
foreach ($lines as $line) {
	printf("INSERT INTO xxx_table (%s) VALUES (%s);\n", $header_s, implode(',', SQLUtil::escapeAll($line)));
}
